homeassistant/components/transmission/sensor.py

```diff
@@ -22,6 +28,70 @@
 )
 
 
+@dataclass
+class TransmissionSensorEntityDescription(SensorEntityDescription):
+    """Describe Transmission sensor entity."""
+
+    name_suffix: str | None = None
+    sub_type: str | None = None
+
+
+SPEED_SENSOR_DESCRIPTIONS = [
+    TransmissionSensorEntityDescription(
+        key="download",
+        name_suffix="Down Speed",
+        sub_type="download",
+        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    TransmissionSensorEntityDescription(
+        key="upload",
+        name_suffix="Up Speed",
+        sub_type="upload",
+        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+]
+STATUS_SENSOR_DESCRIPTIONS = [
+    TransmissionSensorEntityDescription(
+        key="status",
+        name_suffix="Status",
+    ),
+]
+TORRENTS_SENSOR_DESCRIPTIONS = [
+    TransmissionSensorEntityDescription(
+        key="active_torrents",
+        name_suffix="Active Torrents",
+        sub_type="active",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    TransmissionSensorEntityDescription(
+        key="paused_torrents",
+        name_suffix="Paused Torrents",
+        sub_type="paused",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    TransmissionSensorEntityDescription(
+        key="total_torrents",
+        name_suffix="Total Torrents",
+        sub_type="total",
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    TransmissionSensorEntityDescription(
+        key="completed_torrents",
+        name_suffix="Completed Torrents",
+        sub_type="completed",
+        state_class=SensorStateClass.MEASUREMENT,
```

Comment:
No. This sensor shows a number of torrents currently in "completed" (i.e., seeding) state. If user deletes one, count goes down as well. So measurement is the correct state class here.
Same for other sensors.

Reply:
So its a total and not a measurement?

Reply:
Maybe I don't fully understand the purpose of `total`...

> Entities tracking a total amount have a value that may optionally reset periodically, like this month's energy consumption, today's energy production or the yearly growth of a stock portfolio. The sensor's value when the first statistics is compiled is used as the initial zero-point.

This doesn't apply to this sensor because it only shows the current value of a certain attribute and doesn't get reset periodically (or ever really).

Reply:
>  and doesn't get reset periodically (or ever really).

What does that mean? It keeps counting up? Or does it go down too?

Please note that the `total` class **optionally** resets. `total` keeps track of differences between the sums, while `measurement` only tracks a state of point in time (and e.g., tracks the average completed torrents in this case which seems odd).

Reply:
Okay, I probably didn't explain it well enough.
Transmission is a torrent client and it exposes the number of torrent files currently tracked by application. You can add torrents (increasing "Total Torrents" sensor) and you can delete them (reducing it). Torrents can have different states - i.e., started or completed, active or paused. Sensors track number of torrents **currently** having a certain state, they don't count any totals. It's essentially the same as a sensor tracking a number of people currently inside a house - you would probably not use a total state class for that.

> tracks the average completed torrents in this case which seems odd

Actually that would track the average number of torrents that are completed and currently seeding which makes a certain sense. Again, it's not some sort of lifetime total or even daily total, it's just the number of objects in a certain state at this particular moment. :)

Reply:
Maybe it would be less confusing is "Completed Torrents" was called "Seeding Torrents" instead

Reply:
Maybe it would be less confusing if "Completed Torrents" was called "Seeding Torrents" instead